src/d3a/models/strategy/update_frequency.py

```diff
@@ -67,7 +67,12 @@ def _set_or_update_energy_rate_change_per_update(self):
                     (self.initial_rate[slot] - self.final_rate[slot]) / \
                     self.number_of_available_updates
             else:
-                energy_rate_change_per_update[slot] = self.energy_rate_change_per_update[slot]
+                if self.rate_limit_object is min:
```

Comment:
So, to understand the sign convention here, for the case of bids the `energy_rate_change_per_update` is negative, and for the offers is positive, right ? If I understood it correctly, it is a bit counter-intuitive, but ok for me.

Reply:
Yes, you are right. If you have a look at this [method](https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/strategy/update_frequency.py#L87), updated rate is found via subtracting `energy_rate_change_per_update` from `initial_rate`. In case of load, rate has to increase increase so, sign of `energy_rate_change_per_update` has to be reversed. I hope I didn't confused you more

Reply:
I agree with Spyros, that this is counter intuitive . You could also change the sign in your mentioned method, Faizan. But ok, leave it like this for now.